packages/block-editor/src/store/reducer.js

```diff
@@ -91,6 +91,18 @@ function flattenBlocks( blocks, transform = identity ) {
 	return result;
 }
 
+function getFlattenedClientIds( blocks ) {
```

Comment:
If this function is called many times, and is worth a performance improvement, I think avoiding the stack array and the array operations and simply having a recursive function navigating the inner blocks tree is more performant.

Reply:
Actually no, a recursive function is always less performant than the stacked alternative.